app/controllers/users_controller.rb

```diff
@@ -18,22 +26,49 @@ def confirm_email
     end
   end
 
-  def create
+  def create_parent
     unless @user = existing_user(params[:user][:email])
       @user = User.new(user_params)
       @user.role = "parent"
     else
       flash[:alert] = "This email is already in use, please use a different email"
-      redirect_to signup_path
+      redirect_to signup_parent_path
     end
 
     if @user.save
+      puts "USER PARENT SAVED"
       UserMailer.registration_confirmation(@user).deliver
-      flash[:notice] = "Thank you for creating an account. A confirmation email has been sent #{@user.email}. Please click on the link to verify your account."
+      flash[:notice] = "Thank you for creating an account. A confirmation email has been sent to #{@user.email}. Please click on the link to verify your account."
       redirect_to login_path
     else
+      puts "USER PARENT NOT SAVED"
       flash[:alert] = "Oops! Something went wrong, please try again."
-      redirect_to signup_path
+      redirect_to signup_parent_path
+    end
+  end
+
+  def create_director
+    @user= User.new(user_params)
+    @user.role = "director"
+    if @user.save
+      puts "USER SAVED"
+      @director = Director.new(
+        user_id: @user.id
+      )
+      if @director.save
+        puts "DIRECTOR SAVED"
+        UserMailer.registration_confirmation(@user).deliver
+        flash[:notice] = "Director Created! Please ask new director to verify email"
+        redirect_to login_path
+      else
+        puts "DIRECTOR NOT SAVED"
+        flash[:alert] = "Oops! Something went wrong, please try again"
+        redirect_to signup_director_path
+      end
+    else
+      puts "USER NOT SAVED"
+      flash[:alert] = "Oops! Something went wrong, please try again!"
```

Comment:
For the alerts, can you instead put @director.errors.full_messages? Replace all of them please. This will ensure that the user sees the actual error messages